### PR TITLE
docs: Update pages with new version documentation

### PR DESCRIPTION
**Pull request by bog-walk**
Edit docs to be in sync with the updated Wiki pages after the release of version 0.42.0:
- Add new subsections in: Deep Dive into DAO, Deep Dive into DSL
- Edit FAQ and Modules pages
- Add Schemas section to tree with the subsections: Table Definition, Data Types, and Functions
  - matches the new documentation structure and allows addition of new feature sections

### CODE REVIEW

**Review by hhariri — APPROVED** [review]

File: documentation-website/Writerside/topics/Table-Definition.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+* `onDelete: ReferenceOption? = null` is an action to the case when linked row from a parent table can be deleted.
+* `onUpdate: ReferenceOption? = null` is an action to the case when value in a referenced column can be changed.
+* `fkName: String? = null` is a foreign key constraint name.
+
+Enum class `ReferenceOption` has four values:
+
+* `RESTRICT` is an option restricts changes on referenced column, and the default option for MySQL dialect.
+* `NO_ACTION` is the same as RESTRICT, and the default option for Oracle and SQL Server dialects.
+* `CASCADE` is an option allows updating or deleting the referring rows.
+* `SET_NULL` is an option sets null to the referring column values.
```
Comment (joc-a) [inline documentation-website/Writerside/topics/Table-Definition.md:167]:
```suggestion
* `RESTRICT` is an option that restricts changes on a referenced column, and the default option for MySQL dialect.
* `NO_ACTION` is the same as RESTRICT, and the default option for Oracle and SQL Server dialects.
* `CASCADE` is an option that allows updating or deleting the referring rows.
* `SET_NULL` is an option that sets the referring column values to null.
* `SET_DEFAULT` is an option that sets the referring column values to the default value.
```

File: documentation-website/Writerside/topics/Table-Definition.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+```
+
+If any `Cities` row will be deleted, the appropriate `Citizens` row will be deleted too.
+
+### Check
+
+The `CHECK` SQL constraint checks that all values in a column match the condition. Exposed supports the `check()` method. 
+You apply this method to a column and pass the appropriate condition to it.
+
+For example, to check that the `name` column contains strings that begins with capital letter, use the following code:
```
Comment (joc-a) [inline documentation-website/Writerside/topics/Table-Definition.md:186]:
```suggestion
For example, to check that the `name` column contains strings that begin with a capital letter, use the following code:
```

File: documentation-website/Writerside/topics/Table-Definition.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    val name = varchar("name", 50)
+    val city = optReference("city", Cities.name, onDelete=ReferenceOption.CASCADE)
+}
+```
+
+If any `Cities` row will be deleted, the appropriate `Citizens` row will be deleted too.
+
+### Check
+
+The `CHECK` SQL constraint checks that all values in a column match the condition. Exposed supports the `check()` method.
```
Comment (joc-a) [inline documentation-website/Writerside/topics/Table-Definition.md:183]:
```suggestion
The `CHECK` SQL constraint checks that all values in a column match some condition. Exposed supports the `check()` method. 
```

File: documentation-website/Writerside/topics/Table-Definition.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+The `CHECK` SQL constraint checks that all values in a column match the condition. Exposed supports the `check()` method. 
+You apply this method to a column and pass the appropriate condition to it.
+
+For example, to check that the `name` column contains strings that begins with capital letter, use the following code:
+```kotlin
+// SQL: CONSTRAINT check_Cities_0 CHECK (REGEXP_LIKE("NAME", '^[A-Z].*', 'c')))
+val name = varchar("name", 50).check { it regexp "^[A-Z].*" }
+```
+
+Some DBMS could not support any condition. For more information, see its documentations.
```
Comment (joc-a) [inline documentation-website/Writerside/topics/Table-Definition.md:192]: What does this sentence mean? Some databases do not support the CHECK constraint altogether?

File: documentation-website/Writerside/topics/Data-Types.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+such columns in an automatic way, but that doesn't mean that you can't use such column types.
+You have two options to work with ENUM database types:
+1. Use existing ENUM column from your tables
+2. Create column from Exposed by providing raw definition SQL
+   In both cases, you should use `customEnumeration` function (available since version 0.10.3)
+
+As a jdbc-driver can provide/expect specific classes for Enum type, you must provide from/to transformation functions for 
+them when defining a `customEnumeration`.
+
+For such enum `private enum class Foo { Bar, Baz }` you can use provided code for your database:
```
Comment (joc-a) [inline documentation-website/Writerside/topics/Data-Types.md:56]:
```suggestion
For such enum `private enum class Foo { Bar, Baz }`, you can use the provided code for your database:
```

File: documentation-website/Writerside/topics/Data-Types.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+The `exposed-json` extension (`org.jetbrains.exposed:exposed-json:$exposed_version`) provides additional types 
+(see [how to use](#how-to-use-json-and-jsonb-types)):
+
+* `json` - `JSON`
+* `jsonb` - `JSONB`
+
+**Note**: Databases store JSON values either in text or binary format, so Exposed provides two types to account for any potential 
+differences, if they exist, for example:
+- **PostgreSQL**: `json()` maps to `JSON`, while `jsonb()` maps to `JSONB`.
+- **SQLite**: No native JSON type so `json()` maps to TEXT, while `jsonb()` throws.
```
Comment (joc-a) [inline documentation-website/Writerside/topics/Data-Types.md:40]:
```suggestion
- **SQLite**: No native JSON type, so `json()` maps to TEXT, while `jsonb()` throws.
```

File: documentation-website/Writerside/topics/Data-Types.md [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+**MySQL**
+```Kotlin
+val existingEnumColumn = customEnumeration("enumColumn", { value -> Foo.valueOf(value as String) }, { it.name })
+val newEnumColumn = customEnumeration("enumColumn", "ENUM('Bar', 'Baz')", { value -> Foo.valueOf(value as String) }, { it.name })
+```
+
+**PostgreSQL**
+
+PostgreSQL requires that ENUM is defined as a separate type, so you have to create it before creating your table. 
+Also postgresql jdbc driver returns PGobject instances for such values. The full working sample is provided below:
```
Comment (joc-a) [inline documentation-website/Writerside/topics/Data-Types.md:73]:
```suggestion
Also, PostgreSQL JDBC driver returns PGobject instances for such values. The full working sample is provided below:
```

**Review by joc-a — CHANGES_REQUESTED** [review]

File: documentation-website/Writerside/topics/Table-Definition.md (hunk 4 above)
Comment (bog-walk) [reply documentation-website/Writerside/topics/Table-Definition.md]: Based on the context, yes, I assume it was meant to be related to check constraints specifically. I've written something less vague in the recent changes.

**Review by joc-a — APPROVED** [review]

**Pull request merged**